code/modules/food_and_drinks/recipes/soup_mixtures.dm

```diff
@@ -1228,7 +1228,6 @@
 	)
 	results =  list(
 		/datum/reagent/consumable/nutriment/soup/oatmeal = 20,
-		/datum/reagent/consumable/milk = 12,
```

Comment:
For the record the point of this is not
`The extra 12 units of milk comes from nowhere which doesn't make sense`
It is that the recipe does not consume all of the milk, IE, some of it turns into oatmeal some of it does not. 
Not that I'm strongly opinionated about keeping it or not.

Reply:
Sadly That's not how reaction code works. A reaction stops under 3
4 conditions 
1. If the rate of reaction is too low due to insufficient temperature or ph
2. If the amount of product formed(total_step_add) is >= amount to be formed(step_target_vol)
3. If any of the required reagent volume reaches 0
4. by overriding reaction_step() and you decide when to end the reaction.

For your case you want to go for option 4 so you can end the reaction early if we detect 12 units of milk, anyway.
I also agree with you that it's not a big deal, perhaps add that behavior in a future PR